hicommonwealth/commonwealth
# Unify polls routes into controller

## Link to Issue
Closes: #4902

## Description of Changes
- Move polls logic to new polls controller, methods and handlers
- Updates UI to use new routes
- Misc
  - Delete unused editComment route
  - Fixes some imports that used the wrong `ServerError` type


## Test Plan
- API tests were added for the new routes
- Manual CA tested
  - Create a new thread
  - On the thread, create a poll– confirm that UI shows new poll
  - Submit a vote– confirm UI updates state to show vote
  - Refresh the page– confirm UI shows poll and vote
  - Delete the poll– confirm UI shows poll is gone
  - Refresh the page again– confirm UI again shows that poll is gone


## Deployment Plan
N/A

## Other Considerations
- Have not yet removed the old routes, as I want to ensure the new implementations are correct first
- The UI doesn't allow updating a vote, but the API does (both old and new implementation)
